docs/api.rst

```diff
@@ -0,0 +1,219 @@
+.. _ksql-http-api:
+
+HTTP API Reference
+==================
+
+Content Types
+-------------
+
+The ksql HTTP server uses content types for requests and responses to indicate the serialization formato f the data. Currently, the only serialization format supported is JSON, specified as ``application/json``. Your request should specify this content type in the ``Accept`` header::
+
+    Accept: application/json
+
+Errors
+------
+
+All API endpoints use a standard error message format for any requests that return an HTTP status indicating an error (any 4xx or 5xx statuses): 
+
+   .. sourcecode:: http
+
+      HTTP/1.1 <Error Status>
+      Content-Type: application/json 
+
+      {
+          "message": <Error Message>
+      }
+
+Run a KSQL Statement
+--------------------
+
+The ksql resource runs a sequence of ksql statements. Any statement except those starting with ``SELECT`` can be run on this endpoint. To run ``SELECT`` statements use the ``/query`` endpoint.
+
+.. http:post:: /ksql
+
+   Run a sequence of ksql statements.
+
+   :json string ksql: A semicolon-delimited sequence of KSQL statements to run.
+   :json map streamsProperties: Property overrides to run the statements with. Refer to the :ref:`Config Reference <ksql-param-reference>` for details on properties that can be set.
+   :json string streamsProperties[<property-name>]: The value of the property named by property-name. Both the value and property-name should be strings.
+
+   The response json is an array of result objects. The contents of each result object depends on the statement for which it is returning results. The following sections detail the contents of the result objects by statement.
+
+   **CREATE**
+
+   :>json string currentStatus.statementText: The ksql statement for which the result is being returned.
+   :>json string currentStatus.commandStatus: One of QUEUED, PARSING, EXECUTING, RUNNING, TERMINATED, SUCCESS, or ERROR
```

Comment:
is there any way to get this from an show/describe? seems strange we only get status info once, when there are so many states

Reply:
@norwood - there is a status endpoint you can query to get the current status for a command. I was originally just going to document it for the new 5.0 api. But now I think we should add it to this doc as well - a client could get a non-terminal status back and should then have a way to poll the command result. It’s also worth considering making this info available from describe. I’ll file an issue.